integration-test/src/test/java/org/apache/iotdb/db/it/IoTDBConfigNodeIT.java

```diff
@@ -0,0 +1,208 @@
+                == dataNodeLocation.getDataRegionConsensusEndPoint().getPort()) {
+          found = true;
+          break;
+        }
+      }
+      assertTrue(found);
+    }
+  }
+
+  @Test
+  public void removeAndStopConfigNodeTest() throws TException, IOException, InterruptedException {
+    TShowClusterResp clusterNodes;
+    TSStatus status;
+
+    List<ConfigNodeWrapper> configNodeWrappers = EnvFactory.getEnv().getConfigNodeWrapperList();
+    List<DataNodeWrapper> dataNodeWrappers = EnvFactory.getEnv().getDataNodeWrapperList();
+
+    try (SyncConfigNodeIServiceClient client =
+        (SyncConfigNodeIServiceClient) EnvFactory.getEnv().getConfigNodeConnection()) {
+      // add ConfigNode
+      for (int i = 0; i < 2; i++) {
+        ConfigNodeWrapper configNodeWrapper =
+            new ConfigNodeWrapper(
+                false,
+                configNodeWrappers.get(0).getIpAndPortString(),
+                "IoTDBConfigNodeIT",
+                "removeAndStopConfigNodeTest",
+                EnvUtils.searchAvailablePorts());
+        configNodeWrapper.createDir();
+        configNodeWrapper.changeConfig(ConfigFactory.getConfig().getConfignodeProperties());
+        configNodeWrapper.start();
+        configNodeWrappers.add(configNodeWrapper);
+      }
+      EnvFactory.getEnv().setConfigNodeWrapperList(configNodeWrappers);
+      clusterNodes = getClusterNodeInfos(client, 3, 3);
+      assertEquals(TSStatusCode.SUCCESS_STATUS.getStatusCode(), clusterNodes.getStatus().getCode());
+      List<TConfigNodeLocation> configNodeLocationList = clusterNodes.getConfigNodeList();
+      List<TDataNodeLocation> dataNodeLocationList = clusterNodes.getDataNodeList();
+      checkNodeConfig(
+          configNodeLocationList, dataNodeLocationList, configNodeWrappers, dataNodeWrappers);
+
+      // test remove ConfigNode
+      TConfigNodeLocation removedConfigNodeLocation = clusterNodes.getConfigNodeList().get(1);
+      for (int i = 0; i < retryNum; i++) {
+        Thread.sleep(1000);
+        status = client.removeConfigNode(removedConfigNodeLocation);
+        if (TSStatusCode.SUCCESS_STATUS.getStatusCode() == status.getCode()) {
+          break;
+        }
+      }
+
+      clusterNodes = getClusterNodeInfos(client, 2, 3);
+      assertEquals(TSStatusCode.SUCCESS_STATUS.getStatusCode(), clusterNodes.getStatus().getCode());
+      checkNodeConfig(
+          clusterNodes.getConfigNodeList(),
+          clusterNodes.getDataNodeList(),
+          configNodeWrappers,
+          dataNodeWrappers);
+
+      List<TConfigNodeLocation> configNodeList = clusterNodes.getConfigNodeList();
+      for (TConfigNodeLocation configNodeLocation : configNodeList) {
+        assertNotEquals(removedConfigNodeLocation, configNodeLocation);
+      }
+
+      // test stop ConfigNode
+      status = client.stopConfigNode(removedConfigNodeLocation);
+      assertEquals(TSStatusCode.SUCCESS_STATUS.getStatusCode(), status.getCode());
+    }
```

Comment:
Adding a `catch (Exception e) ` dealing logic, 
when an exception throws, this test should be considered as failed.

Reply:
done